tools/ci_build/build.py

```diff
@@ -1110,6 +1117,25 @@ def run_training_python_frontend_e2e_tests(cwd):
     # frontend tests are to be added here:
     log.info("Running python frontend e2e tests.")
 
+    def mpirun_subprocess(mpicmd, cwd):
+        log.debug("Running mpi subprocess in '{0}'\n{1}".format(cwd or os.getcwd(), mpicmd))
+        mpi = subprocess.Popen(
+            mpicmd,
+            cwd=cwd,
+            shell=True)
+        mpi.wait()
+        if mpi.returncode != 0:
+            raise TestFailure(mpicmd, "Test failed with returncode={}.".format(mpi.returncode))
+
+        log.debug("MPI subprocess completed. Return code=" + str(mpi.returncode))
+
+    import torch
+    ngpus = torch.cuda.device_count()
+    if ngpus > 1:
+        run_glue_mpicmd = 'mpirun -n {} {} orttraining_run_glue.py'.format(ngpus, sys.executable)
+        log.debug("RUN: " + run_glue_mpicmd)
+        mpirun_subprocess(run_glue_mpicmd, cwd)
```

Comment:
can run_subprocess() be used here?

Reply:
MPI run calls 'mpirun' with 'python' as an argument. it needs to wait for all parallel processes to complete. these are things different from run_subprocess(). These differences are significant enough so for the need of mpirun_subprocess().

Reply:
i'm not sure i see the reason for a separate function. run_subprocess can be made more general if needed.
as far as i know it should be able to handle ["mpirun", sys.executable, ...] and as long as mpirun returns after all child processes complete, wouldn't it work with subprocess.run() too?